### PR TITLE
Fix Doxygen warnings for anchors

### PR DESCRIPTION
**Pull request by jpthiele**
This fixes the anchors in news, step and (auto generated) gallery files,
such that Doxygen does not throw 'multiple use of section label' warnings.

### CODE REVIEW

**Comment by jpthiele** [comment]
This does some of the same changes as #16048, so it is marked as Draft for now.
 If the above PR is merged into master I will merge master in here and squash.

**Review by masterleinad — APPROVED** [review]

**Comment by bangerth** [comment]
Can you rebase to get rid of the merge-conflicts?

File: doc/doxygen/scripts/make_gallery.pl [hunk 1]
```diff
@@ -163,7 +163,7 @@
         print "<a name=\"ann-$file\"></a>\n";
         print "<h1>Annotated version of $file</h1>\n";
 
-        system $^X, "$cmake_source_dir/doc/doxygen/scripts/program2doxygen", "$gallery_dir/$file";
+        system $^X, "$cmake_source_dir/doc/doxygen/scripts/program2doxygen.pl", "$gallery_dir/$file" , "--prefix=$file";
```
Comment (bangerth) [inline doc/doxygen/scripts/make_gallery.pl:166]: You use the name of the file as a prefix, but does that always lead to allowed characters in an HTML anchor?

File: examples/step-1/doc/intro.dox [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<a name="Intro"></a>
+<a name="step-1-Intro"></a>
```
Comment (bangerth) [inline examples/step-1/doc/intro.dox:1]: For all of these anchors in the tutorial programs, have you searched whether there are references to these names? As in `<a href="..."`?

Comment (jpthiele) [reply examples/step-1/doc/intro.dox:1]: For the intro and result files this is done in `intro2toc.pl` and I added a prefix parameter to it as well.

File: doc/doxygen/scripts/make_gallery.pl (hunk 1 above)
Comment (jpthiele) [reply doc/doxygen/scripts/make_gallery.pl:166]: In line 163 it is used as well so I supposed that it would be fine.

**Comment by jpthiele** [comment]
Was closed automatically when force pushing the reset and fast forwarded branch (since it was the same as the master branch)

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
Thank you!

**Pull request merged**